File: C++/프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.cpp
#include <algorithm>
#include <vector>
#include <unordered_map>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> data, string ext, int val_ext, string sort_by) {
    vector<vector<int>> answer;
    unordered_map<string, int> word_map;
    word_map.insert({"code", 0});
    word_map.insert({"date", 1});
    word_map.insert({"maximum", 2});
    word_map.insert({"remain", 3});
    
    for(auto& d : data) {
        if(d[word_map[ext]] < val_ext) {
            answer.push_back(d);
        }
    }
    
    sort(answer.begin(), answer.end(), [&](vector<int> a, vector<int> b) {
        return a[word_map[sort_by]] < b[word_map[sort_by]];
    });
    return answer;
}

File: Python3/프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.py
def solution(data, ext, val_ext, sort_by):
    answer = [[]]
    word_dict = {"code": 0, "date": 1, "maximum": 2, "remain": 3}
    return sorted([d for d in data if val_ext > d[word_dict[ext]]], key=lambda x: x[word_dict[sort_by]])